backend/src/core/cloudstorage.py

```diff
@@ -0,0 +1,125 @@
+import io
+import logging
+import os
+from mimetypes import guess_type
+from typing import BinaryIO, Union
+
+from minio import Minio, S3Error
+from minio.datatypes import Bucket, Object
+
+logger = logging.getLogger(__name__)
+
+
+class CloudStorage:
+    def __init__(
+        self,
+        api_url: str = None,
+        access_key: str = None,
+        secret_key: str = None,
+        secure: bool = None,
+    ):
+        self.url = api_url or os.getenv("MINIO_HOST_URL")
+        self.access_key = access_key or os.getenv("MINIO_ACCESS_KEY")
+        self.secret_key = secret_key or os.getenv("MINIO_SECRET_KEY")
+        self.secure: bool = secure or os.getenv("MINIO_SECURE", False) == "True"
+        self.client = self._connect_to_cloudstorage()
+
+    def _connect_to_cloudstorage(self) -> Minio:
+        try:
+            client = Minio(
+                endpoint=self.url,
+                access_key=self.access_key,
+                secret_key=self.secret_key,
+                secure=self.secure,
+            )
+            self._check_connection(client)
+            return client
+        except Exception as e:
+            logger.error(f"Failed to connect to MinIO server: {e}")
+            raise ConnectionError(f"Unable to connect to MinIO server: {e}")
+
+    def _check_connection(self, client: Minio) -> None:
+        try:
+            client.list_buckets()
+        except S3Error as e:
+            logger.error(f"MinIO S3 Error: {e}")
+            raise
+        except Exception as e:
+            logger.error(f"General error when checking connection to MinIO: {e}")
```

Comment:
We should convert loggings from f-string to %s. Python's logging system is not really compatible with f-strings and sometims we will miss data